algo/src/main/java/com/hartwig/serve/ServeAlgo.java

```diff
@@ -71,59 +71,28 @@ public ServeAlgo(@NotNull final RefGenomeManager refGenomeManager, @NotNull fina
 
     @NotNull
     public Map<RefGenome, ExtractionResult> run(@NotNull ServeConfig config) throws IOException {
-        List<ExtractionResult> extractions = Lists.newArrayList();
-        if (config.useVicc()) {
-            extractions.add(extractViccKnowledge(config.viccJson(), config.viccSources()));
-        }
-
-        if (config.useIclusion()) {
-            extractions.add(extractIclusionKnowledge(config.iClusionTrialTsv(), config.iClusionFilterTsv()));
-        }
-
-        if (config.useCkbEvidence()) {
-            extractions.add(extractCkbEvidenceKnowledge(config.ckbDir(),
-                    config.ckbBlacklistMolecularProfileTsv(),
-                    config.ckbDrugCurationTsv(),
-                    config.ckbBlacklistEvidenceTsv()));
-        }
-
-        if (config.useCkbTrials()) {
-            extractions.add(extractCkbTrialKnowledge(config.ckbDir(),
-                    config.ckbBlacklistMolecularProfileTsv(),
-                    config.ckbBlacklistTrialTsv()));
-        }
-
-        if (config.useDocm()) {
-            extractions.add(extractDocmKnowledge(config.docmTsv()));
-        }
-
-        if (config.useHartwigCohortHotspots()) {
-            extractions.add(extractHartwigCohortHotspotKnowledge(config.hartwigCohortHotspotTsv(), !config.skipHotspotResolving()));
-        }
-
-        if (config.useHartwigCuratedHotspots()) {
-            extractions.add(extractHartwigCuratedHotspotKnowledge(config.hartwigCuratedHotspotTsv(), !config.skipHotspotResolving()));
-        }
-
-        if (config.useHartwigDriverGenes()) {
-            extractions.add(extractHartwigDriverGeneKnowledge(config.driverGene37Tsv()));
-        }
-
-        if (config.useHartwigCuratedGenes()) {
-            extractions.add(extractHartwigCuratedGeneKnowledge(config.hartwigCuratedGeneTsv()));
-        }
-
-        Map<RefGenome, List<ExtractionResult>> versionedMap = refGenomeManager.makeVersioned(extractions);
-
-        Map<RefGenome, ExtractionResult> refDependentExtractionMap = Maps.newHashMap();
-        for (Map.Entry<RefGenome, List<ExtractionResult>> entry : versionedMap.entrySet()) {
-            refDependentExtractionMap.put(entry.getKey(), ExtractionFunctions.merge(entry.getValue()));
-        }
+        Stream<ExtractionResult> ckbExtractions =
+                (config.useCkbEvidence() || config.useCkbTrials()) ? extractCkbKnowledge(config) : Stream.empty();
```

Comment:
Could we combine this with main config to extraction stream somehow? I think even in it incurs a bit of duplication it will be a little more readable.

Reply:
Is the latest change what you had in mind? The main thing I wanted to achieve here was only reading the CKB entries once.

Reply:
Got it, yeah what you did in the last commit looks good to me!